client/src/connection/rest/session.ts

```diff
@@ -204,6 +204,7 @@ export class ComputeSession extends Compute {
       "warning",
       "completed",
       "idle",
+      "failed",
```

Comment:
failed is one of the possible job states that indicates that the request is finished. Add it here so that the while loop will properly terminate on failed jobs.